stripe-core/src/main/java/com/stripe/android/core/networking/AnalyticsRequestV2Executor.kt

```diff
@@ -0,0 +1,60 @@
+package com.stripe.android.core.networking
+
+import android.app.Application
+import androidx.annotation.RestrictTo
+import androidx.work.Constraints
+import androidx.work.NetworkType
+import androidx.work.OneTimeWorkRequestBuilder
+import androidx.work.WorkManager
+import androidx.work.await
+import com.stripe.android.core.Logger
+import com.stripe.android.core.utils.IsWorkManagerAvailable
+import javax.inject.Inject
+
+@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+fun interface AnalyticsRequestV2Executor {
+    suspend fun enqueue(request: AnalyticsRequestV2)
+}
+
+@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+class DefaultAnalyticsRequestV2Executor @Inject constructor(
+    private val application: Application,
+    private val networkClient: StripeNetworkClient,
+    private val logger: Logger,
+    private val isWorkManagerAvailable: IsWorkManagerAvailable,
+) : AnalyticsRequestV2Executor {
+
+    override suspend fun enqueue(request: AnalyticsRequestV2) {
+        val isEnqueued = isWorkManagerAvailable() && enqueueRequest(request)
+        if (!isEnqueued) {
+            executeRequest(request)
+        }
+    }
+
+    private suspend fun enqueueRequest(request: AnalyticsRequestV2): Boolean {
+        val workManager = WorkManager.getInstance(application)
+        val inputData = SendAnalyticsRequestV2Worker.createInputData(request)
+
+        val constraints = Constraints.Builder()
+            .setRequiredNetworkType(NetworkType.CONNECTED)
+            .setRequiresBatteryNotLow(true)
```

Comment:
Should we maybe allow analytics to be fired even on low battery scenarios?

Reply:
Actually yes, let’s do that